### DISCUSSION TITLE
getauxval - ld-linux with toolchain/appimage box86

### DISCUSSION THREAD

**Discussion by Jacobw1oo** [ptitSeb/box86#834] [Q&A]
For appimage box86, 
Command LD_SHOW_AUXV=1 sleep 1 box86 : Returns Error file not find. and steam needs it.

Aarch64 returns it correctly with ld-linux-aarch64-so.1
Armv7 chroot returns it correctly with ld-linux-armhf.so.3 

is there a way to preload or pass this system info?

**Comment by ptitSeb** [comment]
I'm unsure what you are trying to do here. Is box86 accessible in your PATH?

**Reply by Jacobw1oo** [reply]
yes. box86 has run several programs that are x86, and as i need new libs i add them to the Appimage, it like a stand alone container build from a chroot, but the chroot is not running when box86 is called.  ld-linux is a core function of a system, and box86 is not running on the chroot or an official mutiarch.

**Reply by ptitSeb** [reply]
ok, but I still don't understand what you are trying to do. Are you trying to run x86 `ld-linux.so`?

**Reply by Jacobw1oo** [reply]
Ultimately they to get steam to run. for it error is GLib-Error getauxval()
Try to get the get getauxval() to return data. my box86 has ld-linux.so.
commands tested:
LD_SHOW_AUXV=1 sleep 1 - returns info for aarch64
LD_SHOW_AUXV=1 sleep 1 box86 - returns info for aarch64
box86 LD_SHOW_AUXV=1 sleep 1 - returns Error: file is not found
BOX86_LD_PRELOAD=/chroot/lib/ld-linux-armhf.so.3 box86 steam - returns error is GLib-Error getauxval()
BOX86_LD_PRELOAD=/chroot/lib/ld-linux-armhf.so.3 box86 LD_SHOW_AUXV=1 sleep 1 returns Error: file is not found

**Reply by ptitSeb** [reply]
`LD_SHOW_AUXV=1 sleep ....` will show auval for `sleep`, so the aarch64 program.
I can run steam on my side (no chroot) currently. I still don't understand what you are trying to do.
`LD_SHOW_AUXVAL=1` will always be interpreted by the ld-linux that is used (aarch64 or armhf), so , again, not sure what you are trying to acheive but it will not show the x86/x86_64 one unless you are running in a x86/x86_64 bash session directly (box86/box64 can run bash)

**Reply by Jacobw1oo** [reply]
Here is  full terminal log:
[alarm@danctnix build]$ steam
/usr/local/bin/steam: line 3: /home/alarm/.local/share/Steam/: Is a directory
/home/alarm/.local/share/Steam/steam.sh: line 124: VERSION_ID: unbound variable
/home/alarm/.local/share/Steam/steam.sh: line 124: VERSION_ID: unbound variable
steam.sh[17181]: Running Steam on archarm  64-bit
steam.sh[17181]: STEAM_RUNTIME is enabled by the user
setup.sh[17248]: Steam runtime environment up-to-date!
steam.sh[17181]: Steam client's requirements are satisfied
[2023-06-14 19:34:25] Startup - updater built Jun 14 2023 20:16:17
[2023-06-14 19:34:25] Startup - Steam Client launched with: '/home/alarm/.local/share/Steam/ubuntu12_32/steam' 'steam://open/minigameslist'
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2023-06-14 19:35:00] Loading cached metrics from disk (/home/alarm/.local/share/Steam/package/steam_client_metrics.bin)
[2023-06-14 19:35:00] Using the following download hosts for Public, Realm steamglobal
[2023-06-14 19:35:00] 1. https://client-update.akamai.steamstatic.com, /, Realm 'steamglobal', weight was 1000, source = 'update_hosts_cached.vdf'
[2023-06-14 19:35:00] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 1, source = 'update_hosts_cached.vdf'
[2023-06-14 19:35:00] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[2023-06-14 19:35:01] Checking for update on startup
[2023-06-14 19:35:01] Checking for available updates...
[2023-06-14 19:35:01] Downloading manifest: https://client-update.akamai.steamstatic.com/steam_client_ubuntu12
[2023-06-14 19:35:01] Manifest download: send request
[2023-06-14 19:35:01] Manifest download: waiting for download to finish
[2023-06-14 19:35:02] Manifest download: finished
[2023-06-14 19:35:02] Download skipped: /steam_client_ubuntu12 version 1686779606, installed version 1686779606, existing pending version 0
[2023-06-14 19:35:02] Nothing to do
[2023-06-14 19:35:02] Verifying installation...
[2023-06-14 19:35:02] Performing checksum verification of executable files
[2023-06-14 19:35:05] Verification complete
Error initializing native libpulse.so.0 (last dlerror is libpulsecommon-16.1.so: wrong ELF class: ELFCLASS64)
Error: Symbol innetgr not found, cannot apply R_386_JMP_SLOT 0xf756e118 (0x1c56) in /home/alarm/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libwrap.so.0
Error: Symbol yp_get_default_domain not found, cannot apply R_386_JMP_SLOT 0xf756e180 (0x1df6) in /home/alarm/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libwrap.so.0
Error: Global Symbol gethostid not found, cannot apply R_386_GLOB_DAT @0xe2dd2f80 ((nil)) in /home/alarm/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpulsecommon-1.1.so
Error: Symbol posix_madvise not found, cannot apply R_386_JMP_SLOT 0xe2dd32a4 (0xbcc6) in /home/alarm/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpulsecommon-1.1.so
Error: Symbol personality not found, cannot apply R_386_JMP_SLOT 0xe2dd34bc (0xc526) in /home/alarm/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libpulsecommon-1.1.so
Error: Symbol g_settings_get_double not found, cannot apply R_386_JMP_SLOT 0xe7ea1850 (0x27e2b6) in /home/alarm/.local/share/Steam/ubuntu12_32/steamui.so
Error: Symbol g_settings_new not found, cannot apply R_386_JMP_SLOT 0xe7ea185c (0x27e2e6) in /home/alarm/.local/share/Steam/ubuntu12_32/steamui.so
Loaded SDL version 3.0.0-1735-g2e465ae31
src/common/friendsuihelpers.cpp (50) : Is64BitOS()
src/common/friendsuihelpers.cpp (50) : Is64BitOS()
pcilib: Cannot open /proc/bus/pci
lspci: Cannot find any working access method.
XRRGetOutputInfo Workaround: initialized with override: 0 real: 0x401f00b0
XRRGetCrtcInfo Workaround: initialized with override: 0 real: 0x401f00c0
GetWin32Stats: display was not open yet, good
ComputeStartupMode: found registry default startup mode: 0
Switching to desktopui, since -vgui was not specified
pcilib: Cannot open /proc/bus/pci
lspci: Cannot find any working access method.
GetWin32Stats: display was not open yet, good
src/common/html/chrome_ipc_client.cpp (291) : Is64BitOS()
src/common/html/chrome_ipc_client.cpp (291) : Is64BitOS()
src/common/html/chrome_ipc_client.cpp (291) : Is64BitOS()
src/common/html/chrome_ipc_client.cpp (291) : Is64BitOS()
src/common/html/chrome_ipc_client.cpp (291) : Is64BitOS()
src/common/html/chrome_ipc_client.cpp (291) : Is64BitOS()
CAppInfoCacheReadFromDiskThread took 1500 milliseconds to initialize

(ld-linux-armhf.so.3:17293): GLib-ERROR **: 19:35:36.989: getauxval () failed: No such file or directory
/tmp/.mount_box86t8wvnG/AppRun: line 39: 17293 Trace/breakpoint trap   (core dumped) "${LD_INT}" "${PRELOAD_FLAGS[@]}" "${BOX86_BIN}" "$@"
[alarm@danctnix build]$

**Reply by Hacker1245** [reply]
why are you running this in an appimage?
